src/SharethisManager.php

```diff
@@ -64,4 +67,166 @@ public function getOptions() {
+      $service_array = explode(",", $data_options['services']);
+      $st_spans = "";
+      foreach ($service_array as $service_full) {
+        // Strip the quotes from the element in the array (They are there for javascript).
+        $service = explode(":", $service_full);
+
+        // Service names are expected to be parsed by Name:machine_name. If only one
+        // element in the array is given, it's an invalid service.
+        if (count($service) < 2) {
+          continue;
+        }
+
+        // Find the service code name.
+        $serviceCodeName = Unicode::substr($service[1], 0, -1);
+
+        // Switch the title on a per-service basis if required.
+        // $title = $title;.
+        switch ($serviceCodeName) {
+          case 'twitter':
+            $title = empty($data_options['twitter_suffix']) ? $title : Html::escape($title) . ' ' . Html::escape($data_options['twitter_suffix']);
+            break;
+        }
+
+        // Sanitize the service code for display.
+        $display = Html::escape($serviceCodeName);
+
+        // Put together the span attributes.
+        $attributes = array(
+          'st_url' => $mpath,
+          'st_title' => $title,
+          'class' => 'st_' . $display . $type,
+        );
+        if ($serviceCodeName == 'twitter') {
+          if (!empty($data_options['twitter_handle'])) {
+            $attributes['st_via'] = $data_options['twitter_handle'];
+            $attributes['st_username'] = $data_options['twitter_recommends'];
+          }
+        }
+        // Only show the display text if the type is set.
+        if (!empty($type)) {
+          $attributes['displayText'] = Html::escape($display);
+        }
+        $meta_generator = array(
+          '#type' => 'html_tag',
+          '#tag' => 'span',
+          '#attributes' => $attributes,
+        // It's an empty span tag.
+          '#value' => '',
+        );
+        // Render the span tag.
+        $st_spans .= drupal_render($meta_generator);
+      }
+      $this->sharethis_include_js();
+      return ['data_options' => $data_options, 'm_path' => $mPath, 'm_title' => $mTitle , 'st_spans' => $st_spans];
+
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  function sharethis_include_js() {
+    $has_run = &drupal_static(__FUNCTION__, FALSE);
+    if (!$has_run) {
+      // These are the ShareThis scripts:
+      $data_options = $this->getOptions();
+      $st_js_options = array();
+      $st_js_options['switchTo5x'] = $data_options['widget'] == 'st_multi' ? TRUE : FALSE;
+      if ($data_options['late_load']) {
+        $st_js_options['__st_loadLate'] = TRUE;
+      }
+      $st_js = "";
+      foreach ($st_js_options as $name => $value) {
+        $st_js .= 'var ' . $name . ' = ' . Json::decode($value) . ';';
+
+      }
+      if ((isset($_SERVER['HTTPS']) && strtolower($_SERVER['HTTPS']) == 'on') || (isset($_SERVER['HTTP_X_FORWARDED_PROTO']) && strtolower($_SERVER['HTTP_X_FORWARDED_PROTO']) == 'https')) {
```

Comment:
Don't stick with the $_SERVER['HTTPS'] and don't trust $_SERVER it can give us security issues. load the file from //:ws.sharethis.com it will pick the protocol
See for reference https://github.com/naveenvalecha/twitter_block/blob/8.x-1.x/twitter_block.libraries.yml